### PR TITLE
fix(deps)!: upgrade dependencies

### PR DESCRIPTION
**Pull request by Yelinz**
BREAKING CHANGE: Drop support for django < 4.2 and python < 3.10

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -16,9 +16,9 @@ updates:
       timezone: "Europe/Zurich"
     open-pull-requests-limit: 10
     ignore:
-      - dependency-name: django
+      - dependency-name: python-semantic-release
         versions:
-          - ">=4"
+          - ">=7"
```
Comment (open-dynaMIX) [inline .github/dependabot.yml:21]: Should we add something like this?

```
  ignore:
  - dependency-name: django
    versions:
    - ">=4.3"
  ```

**Review by open-dynaMIX — CHANGES_REQUESTED** [review]
Otherwise LGTM :+1:

File: .github/dependabot.yml (hunk 1 above)
Comment (Yelinz) [reply .github/dependabot.yml:21]: No deprecations, so it should be ok to have the django 5 update queued by dependabot and merge when ready.

Comment (anehx) [reply .github/dependabot.yml:21]: @Yelinz I strongly disagree, we always want LTS versions..

Comment (Yelinz) [reply .github/dependabot.yml]: added ignore

**Comment by czosel** [comment]
@Yelinz I missed this MR when I did #270 - I rebased this one, let's get the in before releasing 0.7.0 (#274).

**Pull request merged**